### PR TITLE
Dont trust the problem for the dimension in NS Rhie Chow User Object

### LINKED ISSUE #25936: WCNSFV Rhie Chow user object should not retrieve its dimension from the problem
## Bug Description
WCNSFV can be 2D or 1D in a mixed-dimensional 3D problem

## Steps to Reproduce
Do mixed dimensions NSFV simulations

## Impact
Low once identified, but errors out

### PR DESCRIPTION
**Pull request by GiudGiud**
- on components, problem says 3D even for 2D closes #25936

Be more safe on retrieving the max dimension from the BlockRestrictable interface

### CODE REVIEW

File: framework/include/interfaces/BlockRestrictable.h [hunk 1]
```diff
@@ -104,14 +104,19 @@ class BlockRestrictable
...
-   * @return a set of SudomainIDs that are valid for this object
+   * @return a set of SubdomainIDs that are valid for this object
    */
   const virtual std::set<SubdomainID> & blockIDs() const;
 
   /**
    * Return the largest mesh dimension of the elements in the blocks for this object
    */
-  unsigned int blocksMaxDimension() const { return _blk_dim; }
+  unsigned int blocksMaxDimension() const
```
Comment (lindsayad) [inline framework/include/interfaces/BlockRestrictable.h:114]: move out of the class definition

File: framework/include/interfaces/BlockRestrictable.h [hunk 2]
```diff
@@ -104,14 +104,19 @@ class BlockRestrictable
...
+  {
+    mooseAssert(_blk_dim != std::numeric_limits<unsigned int>::max(),
```
Comment (lindsayad) [inline framework/include/interfaces/BlockRestrictable.h:116]:
```suggestion
    mooseAssert(_blk_dim != libMesh::invalid_uint,
```

File: framework/src/interfaces/BlockRestrictable.C [hunk 3]
```diff
@@ -46,7 +46,8 @@ BlockRestrictable::BlockRestrictable(const MooseObject * moose_object, bool init
                                                   : NULL),
     _boundary_ids(_empty_boundary_ids),
     _blk_tid(moose_object->isParamValid("_tid") ? moose_object->getParam<THREAD_ID>("_tid") : 0),
-    _blk_name(moose_object->getParam<std::string>("_object_name"))
+    _blk_name(moose_object->getParam<std::string>("_object_name")),
+    _blk_dim(std::numeric_limits<unsigned int>::max())
```
Comment (lindsayad) [inline framework/src/interfaces/BlockRestrictable.C:50]:
```suggestion
    _blk_dim(libMesh::invalid_uint)
```

File: framework/src/interfaces/BlockRestrictable.C [hunk 4]
```diff
@@ -63,7 +64,8 @@ BlockRestrictable::BlockRestrictable(const MooseObject * moose_object,
                                                   : NULL),
     _boundary_ids(boundary_ids),
     _blk_tid(moose_object->isParamValid("_tid") ? moose_object->getParam<THREAD_ID>("_tid") : 0),
-    _blk_name(moose_object->getParam<std::string>("_object_name"))
+    _blk_name(moose_object->getParam<std::string>("_object_name")),
+    _blk_dim(std::numeric_limits<unsigned int>::max())
```
Comment (lindsayad) [inline framework/src/interfaces/BlockRestrictable.C:68]:
```suggestion
    _blk_dim(libMesh::invalid_uint)
```

**Review by lindsayad — CHANGES_REQUESTED** [review]

**Review by lindsayad — APPROVED** [review]

File: framework/include/interfaces/BlockRestrictable.h (hunk 1 above)
Comment (GiudGiud) [reply framework/include/interfaces/BlockRestrictable.h]: done

**Pull request merged**